hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractMetadataPopupLayout.java

```diff
@@ -239,7 +239,7 @@ private TextField createKeyTextField() {
         final TextField keyField = new TextFieldBuilder().caption(i18n.getMessage("textfield.key")).required(true)
                 .prompt(i18n.getMessage("textfield.key")).immediate(true)
                 .id(UIComponentIdProvider.METADATA_KEY_FIELD_ID).maxLengthAllowed(MetaData.KEY_MAX_SIZE)
-                .buildTextComponent();
+                .validator(new EmptyStringValidator(i18n.getMessage("textfield.min.length"))).buildTextComponent();
```

Comment:
Define constants for the message keys... the key "textfield.min.length", for example, is also used in other classes (e.g. AbstractCreateUpdateTagLayout)